src/dns/nm-dns-manager.c

```diff
@@ -1637,28 +1637,25 @@ _resolvconf_resolved_managed (void)
 		"/run/systemd/resolve/resolv.conf",
 		"/lib/systemd/resolv.conf",
 		"/usr/lib/systemd/resolv.conf",
+		NULL
 	};
-	GFile *f;
-	GFileInfo *info;
-	gboolean ret = FALSE;
+	struct stat etc_resconf_st;
+	const char* p;
 
-	f = g_file_new_for_path (_PATH_RESCONF);
-	info = g_file_query_info (f,
-	                          G_FILE_ATTRIBUTE_STANDARD_IS_SYMLINK","\
-	                          G_FILE_ATTRIBUTE_STANDARD_SYMLINK_TARGET,
-	                          G_FILE_QUERY_INFO_NOFOLLOW_SYMLINKS,
-	                          NULL, NULL);
+	if (stat (_PATH_RESCONF, &etc_resconf_st) != 0)
+		return FALSE;
 
-	if (info && g_file_info_get_is_symlink (info)) {
-		ret = nm_utils_strv_find_first ((gchar **) resolved_paths,
-		                                G_N_ELEMENTS (resolved_paths),
-		                                g_file_info_get_symlink_target (info)) >= 0;
-	}
+	for (p = resolved_paths[0]; *p; p++) {
+		struct stat systemd_resconf_st;
 
-	g_clear_object(&info);
-	g_clear_object(&f);
+		if (stat (p, &systemd_resconf_st) != 0)
+			continue;
```

Comment:
I wouldn't do that... "After" delays startup, for a very obscure case (yes, I know: it delays it only if resolved is actually enabled).

The entire _resolvconf_resolved_managed() is a heuristic to determine whether to use resolved. The user already can bypass that by configuring the "dns" plugin explicitly.
The heuristic from https://github.com/NetworkManager/NetworkManager/pull/16#discussion_r103333709 should work fairly well.